### PR TITLE
fix(type-decorations): disable in style blocks

### PR DESCRIPTION
**Pull request by Ilanaya**

### CODE REVIEW

File: src/features/typeDecorations.ts [hunk 1]
```diff
@@ -12,6 +13,27 @@ export default () => {
         },
         rangeBehavior: vscode.DecorationRangeBehavior.ClosedClosed,
     })
+
+    const checkIfInStyles = async (document: vscode.TextDocument, position: vscode.Position) => {
+        const {languageId, uri} = document
+        const stylesLangs = new Set(['scss', 'css', 'less', 'sass'])
+        console.log('stylesLangs.includes(languageId)', stylesLangs.has(languageId))
```
Comment (zardoy) [inline src/features/typeDecorations.ts]: Unnecessary logging

**Comment by zardoy** [comment]
Are you sure we should disable it in styles? It still can give useful piece of information such as syntax hint. 
image

The problem here is that we're printing raw markdown instead of *rendered* version: https://github.com/microsoft/vscode/issues/147530 (e.g. we shouldn't use [`.value`](https://github.com/zardoy/vscode-experiments/blob/2ce643202ae7ac688ad9013c668a5e278efdd7f8/src/features/typeDecorations.ts#L58))

**Comment by Ilanaya** [comment]
Removing `.value` doesn't help as `content` in this case evaluates in `l {}` 
image

> Are you sure we should disable it in styles? It still can give useful piece of information such as syntax hint. image
> 
> The problem here is that we're printing raw markdown instead of _rendered_ version: [microsoft/vscode#147530](https://github.com/microsoft/vscode/issues/147530) (e.g. we shouldn't use [`.value`](https://github.com/zardoy/vscode-experiments/blob/2ce643202ae7ac688ad9013c668a5e278efdd7f8/src/features/typeDecorations.ts#L58))

`vscode.MarkdownString` and `vscode.MarkedString` [escape](https://github.com/microsoft/vscode/blob/6d2920473c6f13759c978dd89104c4270a83422d/src/vs/base/browser/markdownRenderer.ts#L296) their values under the hood, so I had to use lodash's `unescape` (I didn't find equivalent in `rambda`) method in order to produce human-readable string

**Comment by zardoy** [comment]
I'll look into this in a few hours.

@ILScc by the way how tailwind in css is going?

**Comment by zardoy** [comment]
My thoughts on this for now:

We should either disable them in style blocks completely out of the box or implement the setting with the value disabled by default. You can do this on your own, I won't care. In case of impl setting, you can leave lodash method..

@ILScc just like with renameVariableParts you need to pick latest commits from main to fix CI (or rebase; you don't need to supress warning as they don't affect CI).

File: src/features/typeDecorations.ts [hunk 2]
```diff
@@ -38,9 +59,10 @@ export default () => {
                     return content
                 })
                 .join('')
-            const typeMatch = /: (.+)/.exec(hoverString)
+            // eslint-disable-next-line no-await-in-loop
+            const typeMatch = (await checkIfInStyles(document, pos)) ? /Syntax: (.*)/.exec(hoverString) : /: (.+)/.exec(hoverString)
```
Comment (zardoy) [inline src/features/typeDecorations.ts]: @ILScc why did you supress `no-await-in-loop`? You actually could actually done this check before loop. This is not optimal.

**Review by zardoy — CHANGES_REQUESTED** [review]

File: src/features/typeDecorations.ts [hunk 3]
```diff
@@ -21,26 +45,27 @@ export default () => {
...
                 .map(content => {
                     if (typeof content === 'object') return content.value
                     return content
                 })
                 .join('')
-            const typeMatch = /: (.+)/.exec(hoverString)
+            const typeMatch = isInStyles ? /Syntax: (.*)/.exec(hoverString) : /: (.+)/.exec(hoverString)
             if (!typeMatch) continue
-            typeString = typeMatch[1]!
+            typeString = unescape(typeMatch[1]!).replaceAll('\\', '')
```
Comment (Ilanaya) [inline src/features/typeDecorations.ts]: Added this to remove annoying `\` in output 
image

File: src/features/typeDecorations.ts [hunk 4]
```diff
@@ -12,6 +14,28 @@ export default () => {
         },
         rangeBehavior: vscode.DecorationRangeBehavior.ClosedClosed,
     })
+
+    const checkIfInStyles = async (document: vscode.TextDocument, position: vscode.Position) => {
+        if (!getExtensionSetting('typeDecorationsInStyles.enable')) return false
```
Comment (Ilanaya) [inline src/features/typeDecorations.ts]: I understand the solution insn't efficient, but right now I just wanted it to work :)

**Comment by zardoy** [comment]
Wait a second, I'll check it

**Comment by zardoy** [comment]
@ILScc can you please check 2 new commits and merge if the code is fine. I already checked the behavior. Some notes:

- From [Contribution Point guides](https://code.visualstudio.com/api/references/contribution-points) we need to follow the structure of settings:
`category.someThing` so I renamed it to `typeDecorations.enableInStyles` to match category.
- I already [stated above](https://github.com/zardoy/vscode-experiments/pull/7#issuecomment-1156380997) that we need to transform MarkDown syntax text into "rendered text". unescape had nothing in common with MarkDown so I replaced it with appropriate lib (you can check lib description to better understand the intention)

**Pull request merged**